src/configure/configurers/configurerFactory.ts

```diff
@@ -0,0 +1,18 @@
+import { Configurer } from './configurerBase';
+import { GitHubWorkflowConfigurer } from '../configurers/githubWorkflowConfigurer';
+import { AzurePipelineConfigurer } from './azurePipelineConfigurer';
+import { GitRepositoryParameters, RepositoryProvider, AzureSession } from '../model/models';
+import { Messages } from '../resources/messages';
+
+export class ConfigurerFactory {
+    public static GetConfigurer(sourceRepositoryDetails: GitRepositoryParameters, azureSession: AzureSession, subscriptionId: string): Configurer {
```

Comment:
Called these out explicitly so that the caller function knows which are required vs optional properties. Having them in signature helps in clearly calling the requirements out.